core/src/main/scala/org/dbpedia/spotlight/db/SpotlightModel.scala

```diff
@@ -125,20 +130,50 @@ object SpotlightModel {
         tokenTypeStore
       ).asInstanceOf[TextTokenizer]
 
-      createTokenizer()
+      if(cores.size == 1)
+        createTokenizer()
+      else
+        new TokenizerWrapper(cores.map(_ => createTokenizer())).asInstanceOf[TextTokenizer]
 
     } else {
       val locale = properties.getProperty("locale").split("_")
       new LanguageIndependentTokenizer(stopwords, stemmer(), new Locale(locale(0), locale(1)), tokenTypeStore)
     }
 
+    val mixture = if(new File(modelFolder, "ranklib-model.txt").exists()) {
+      val weightsLineElements = scala.io.Source.fromFile(new File(modelFolder, "ranklib-model.txt")).getLines().toArray.last.split(" ")
+      val weights = weightsLineElements.map { (elem: String) =>
+        elem.substring(2).toDouble
+      }
+      if (weights.length == featureNames.length) {
+        val featuresAndWeights: List[(String, Double)] = featureNames.zip(weights)
+
+        SpotlightLog.info(this.getClass, "Using LLM weights: %s".format(
+          featuresAndWeights.map {
+            case (name, weight) =>
+              "%s=%s".format(name, weight)
+          }.reduce(_ + ", " + _)
+        ))
+
+        new LogLinearFeatureMixture(
+          featuresAndWeights
+        )
+      }else{
```

Comment:
I assume you mean this part:
```
      }else{
        SpotlightLog.warn(this.getClass, "Could not load LLM weights, wrong number of weights in file! Using UnweightedMixture.")
        new UnweightedMixture(featureNames.toSet)
      }
    } else{
      SpotlightLog.info(this.getClass, "No weights found - using unweighted mixture.")
      new UnweightedMixture(featureNames.toSet)
    }
```

it's just meant to give more detailed error messages. Do you think this needs to be changed?